daffodil-core/src/main/scala/org/apache/daffodil/dsom/Facets.scala

```diff
@@ -129,10 +131,45 @@ trait Facets { self: Restriction =>
   final lazy val patternValues: Seq[FacetValueR] = {
     val values = combinedBaseFacets.filter { case (f, _) => f == Facet.pattern }
     if (values.size > 0) {
-      val res: Seq[FacetValueR] = values.map { case (f, v) => (f, v.r) }
+      val res: Seq[FacetValueR] = values.map {
+        case (f, v) => {
+          //
+          // The DFDL Infoset can contain strings which hold characters
+          // that are not allowed in XML at all.
+          //
+          // In order to talk about these characters in a XSD pattern facet
+          // we use a remapping of such characters into the Unicode
+          // Private Use Area, so as to have XML-legal characters.
+          //
+          // See the section titled "XML Illegal Characters" on this web page:
+          // https://daffodil.apache.org/infoset/
+          //
+          // Before processing a regex of these characters in Daffodil's pattern facet
+          // validation, we must remap these PUA characters back to the originally
+          // intended code points, since that's what the Infoset strings will contain.
+          //
+          // Consider the character code 0xB. This is illegal in XML v1.0 documents.
+          // A DFDL Schema is an XML Schema, which is an XML document; hence, we cannot
+          // use the character with code 0xB directly, nor can we use an XML numeric
+          // character entity like &#xB; for it. The character is simply disallowed in
+          // XML, including DFDL schemas. Hence, we mention 0xB by using a remapping of it
+          // to the PUA area character 0XE00B, which we express by &#xE00B;
+          // Hence a pattern regex like "[&#xE00B;&#x20;0-9a-zA-Z]" will match
+          // the character with char code 0xB (remapped from E00B), as well as spaces,
+          // and alphanumeric characters.
```

Comment:
Just so I'm clear, ``&#x0B;`` is disallowed in an XSD pattern regex because XSD is just a special XML. And based on Daffodil's PUA handling, one must instead use ``&xE00B;``? Then this transforms that into a regex that matches 0x0B? And our internal infoset value that we check this pattern against hasn't been mapped to PUA at the time we run this regex, because we only transform infoset values to PUA right before converting to XML with an XML infoset outputter?

Reply:
Yeah, it's a bit odd, but in any XML context, including the pattern regex, if you want to talk about an XML-illegal character you have to use the PUA character instead. A DFDL schema is an XML document, so the pattern facet regex has to do this also.